example/app/AppComponents.scala

```diff
@@ -0,0 +1,32 @@
+import com.gu.googleauth.{AuthAction, GoogleAuthConfig}
+import controllers.{Application, Login, routes}
+import play.api.libs.ws.ahc.AhcWSComponents
+import play.api.mvc.AnyContent
+import play.api.routing.Router
+import play.api.{ApplicationLoader, BuiltInComponentsFromContext}
+import play.filters.HttpFiltersComponents
+import router.Routes
+
+class AppComponents(context: ApplicationLoader.Context)
+  extends BuiltInComponentsFromContext(context)
+  with AhcWSComponents
+  with HttpFiltersComponents{
+
+  val clientId: String = configuration.get[String]("your.clientId.config.path")
+  val clientSecret: String = configuration.get[String]("your.clientSecret.config.path")
+  val redirectUrl: String = configuration.get[String]("your.redirectUrl.config.path")
+  val domain: String = configuration.get[String]("your.apps-domain.config.path")
+  val googleAuthConfig = GoogleAuthConfig(clientId, clientSecret, redirectUrl, domain)
+
+  val authAction = new AuthAction[AnyContent](googleAuthConfig, routes.Login.loginAction())(
```

Comment:
I feel like this will be the same every time, and should be provided by the library. Previously it was provided via the "AuthActions" trait. While I would generally applaud removing traits we're kind of locked into inheritance anyway with Play. Having `AuthAction` be provided the same way as `Action` made at least a little bit of sense.

An AuthAction (capitalised for consistency with Play and all its docs) that can be configured away from `AnyContent` on a per-action basis and optionally takes a body parser feels like something this new version lacks. This library is used for APIs as well as websites, so I don't think changing content types and parsers is something that we can ignore.